### PR TITLE
Add scopes support and example

### PR DESCRIPTION
**Pull request by oyarzun**
GitLab now supports the following scopes:

screen shot 2017-10-31 at 11 54 07 am

### CODE REVIEW

**Comment by curry684** [comment]
Thanks, I'll try to review asap!

File: test/src/Provider/GitlabTest.php [hunk 1]
```diff
@@ -65,7 +65,7 @@ public function testScopes()
 
         $url = $this->provider->getAuthorizationUrl($options);
 
-        $this->assertContains(urlencode(implode(',', $options['scope'])), $url);
+        $this->assertContains(rawurlencode(implode(' ', $options['scope'])), $url);
```
Comment (curry684) [inline test/src/Provider/GitlabTest.php]: Isn't this supposed to call that new `getScopeSeparator` call?

File: src/Provider/Gitlab.php [hunk 2]
```diff
@@ -82,18 +82,29 @@ public function getResourceOwnerDetailsUrl(AccessToken $token)
...
+     * GitLab uses a space to separate scopes
+     */
+    protected function getScopeSeparator()
+    {
+        return ' ';
+    }
+
+
+
+    /**
```
Comment (curry684) [inline src/Provider/Gitlab.php]: Given that it's a GitLab constant in a GitLab class I'd prefer it to be a true `CONST` declaration on top of the class.

File: README.md [hunk 3]
```diff
@@ -74,13 +74,28 @@ if (!isset($_GET['code'])) {
...
+
+
 ### Performing API calls
 
 Install [`m4tthumphrey/php-gitlab-api`](https://packagist.org/packages/m4tthumphrey/php-gitlab-api) to interact with the
 Gitlab API after authentication. Either connect manually:
 
 ```php
-$client = new \Gitlab\Client('https://my.gitlab.url/api/v3/');
+$client = new \Gitlab\Client('https://my.gitlab.url/api/v4/');
```
Comment (curry684) [inline README.md:98]: Good catch   😉

**Review by curry684 — CHANGES_REQUESTED** [review]

File: test/src/Provider/GitlabTest.php (hunk 1 above)
Comment (curry684) [reply test/src/Provider/GitlabTest.php]: Or the more preferable constant.

Comment (oyarzun) [reply test/src/Provider/GitlabTest.php]: I've pushed a new commit that uses a constant.

**Comment by curry684** [comment]
And [released](https://github.com/omines/oauth2-gitlab/releases/tag/3.1.0), thanks!

**Pull request merged**